zerver/lib/user_groups.py

```diff
@@ -128,6 +171,39 @@ def get_recursive_membership_groups(user_profile: UserProfile) -> "QuerySet[User
     return cte.join(UserGroup, id=cte.col.id).with_cte(cte)
 
 
+def is_user_in_group(
+    user_group: UserGroup, user: UserProfile, *, direct_member_only: bool = False
+) -> bool:
+    if direct_member_only:
+        return UserGroupMembership.objects.filter(user_group=user_group, user_profile=user).exists()
+
+    return get_recursive_group_members(user_group=user_group).filter(id=user.id).exists()
+
+
+def get_user_group_member_ids(
+    user_group: UserGroup, *, direct_member_only: bool = False
+) -> List[int]:
+    if direct_member_only:
+        member_ids = get_user_group_direct_members(user_group)
```

Comment:
This should be a follow-up to avoid excess complexity, but we should probably rename this function to `get_user_group_direct_member_ids`, to avoid it being parallel to `get_recursive_group_members`, which returns a QuerySet.